### PR TITLE
Make fit() support multiple eval_metric

### PR DESCRIPTION
**Pull request by shengwa**
Modified the `model.py` and `callback.py` to support multiple eval_metric when calling `fit()` in #1531.

### CODE REVIEW

**Comment by jsusskin** [comment]
I'm hoping this gets added soon as well. For now, without modifying the model code itself, my workaround is to explicitly loop through training epoch by epoch and report performance on various metrics at the end of each epoch.

**Comment by shengwa** [comment]
@jsusskin Thanks for your reply. I am trying to fix the check failure asap.

> tests/cpp/threaded_engine_test.cc:198:
>  IO operator pushed, should wait for 2 seconds.
> The fox says 42
> No output has been received in the last 10 minutes, this potentially indicates a stalled build or something wrong with the build itself.

@tqchen Do you have any ideas why this happened after I merged another branch having only .md modification?

**Comment by RishabGargeya** [comment]
+1 for this PR. Hoping it can be added soon as an easy way to monitor and log multiple metrics.

File: python/mxnet/model.py [hunk 1]
```diff
@@ -241,7 +245,11 @@ def _train_multi_device(symbol, ctx, arg_names, param_names, aux_names,
                     monitor.toc_print()
 
                 # evaluate at end, so we can lazy copy
-                executor_manager.update_metric(eval_metric, data_batch.label)
+                if not isinstance(eval_metric, list):
+                    executor_manager.update_metric(eval_metric, data_batch.label)
+                else:
+                    for i in range(len(eval_metric)):
```
Comment (tqchen) [inline python/mxnet/model.py]: for i, metric in enumerate(eval_metric)

File: python/mxnet/model.py [hunk 2]
```diff
@@ -212,7 +212,11 @@ def _train_multi_device(symbol, ctx, arg_names, param_names, aux_names,
     for epoch in range(begin_epoch, end_epoch):
         # Training phase
         tic = time.time()
```
Comment (tqchen) [inline python/mxnet/model.py]: I would recommend always make eval_metric a list, and do special transformation if it is not in the beginning

**Comment by tqchen** [comment]
Thanks for the PR. I have made two comments:

- Use enumerate to enumerate over index and item if you need index. other wise, use 
```for metric in eval_metric```

- Always assume the eval_metric is a list, in the beginning of train function, do  conversion if it is not

**Comment by futurely** [comment]
I wasn't aware this PR before opening https://github.com/dmlc/mxnet/pull/1633 which used the composite design patten to simplify the API.

**Comment by futurely** [comment]
Directed here by https://github.com/dmlc/mxnet/issues/1567#issuecomment-196757098.

**Comment by tqchen** [comment]
closing this as #1633 seems was better choice

**Pull request closed without merging**